public/app/plugins/datasource/cloudwatch/datasource.js

```diff
@@ -80,23 +80,28 @@ function (angular, _, moment, dateMath, kbn, templatingVariable, CloudWatchAnnot
       var period;
       var range = end - start;
 
-      var daySec = 60 * 60 * 24;
+      var hourSec = 60 * 60;
+      var daySec = hourSec * 24;
       var periodUnit = 60;
-      if (now - start > (daySec * 15)) { // until 63 days ago
-        periodUnit = period = 60 * 5;
-      } else if (now - start > (daySec * 63)) { // until 455 days ago
-        periodUnit = period = 60 * 60;
-      } else if (now - start > (daySec * 455)) { // over 455 days, should return error, but try to long period
-        periodUnit = period = 60 * 60;
-      } else if (!target.period) {
-        period = (query.namespace === 'AWS/EC2') ? 300 : 60;
-      } else if (/^\d+$/.test(target.period)) {
-        period = parseInt(target.period, 10);
+      if (!target.period) {
```

Comment:
sorry, stupid comment, it was already in a separate function :) did not see it in the diff, though it was all in the query method :)